com.dynatrace.eclipseintegration.plugin/src/com/dynatrace/diagnostics/launcher/LaunchTerminated.java

```diff
@@ -157,11 +157,21 @@ private long timeoutMs() {
 
 	private final TestRunRecorder.ContainsLaunchTestCounts launchTestCounts = new TestRunRecorder.ContainsLaunchTestCounts() {
 		@Override
-		public int expectedTestCountsFor(ILaunch launch) {
-			for (WeakReference<ILaunch> iLaunchWeakReference : JUnitLaunch_testCount.keySet()) {
-				if (launch.equals(iLaunchWeakReference.get())) {
-					return JUnitLaunch_testCount.get(iLaunchWeakReference);
+		public int expectedTestCountsFor(ILaunch launch) throws InterruptedException {
+			long startPolling = System.currentTimeMillis();
+			while (System.currentTimeMillis() < startPolling + 30*1000 ) {
+				// Required data arrives from a different thread. I expect the data
```

Comment:
use indirect form in sentences; treat the code as public, so if you need to leave a comment make it impersonal